src/tracer.js

```diff
@@ -0,0 +1,130 @@
+'use strict'
+
+import BasicSpan from './span'
+import {TextMapPropagator, BinaryPropagator} from './propagation'
+import Sampler from './sampler'
+import Recorder from './recorder'
+
+export default class BasicTracer {
+    constructor({sampler, recorder} = {}) {
```

Comment:
please do include comments for the constructor and the expected types of the sampler and recorder